stellargraph/mapper/sequences.py

```diff
@@ -366,11 +366,13 @@ class FullBatchSequence(Sequence):
             where C is the target size (e.g., number of classes for one-hot class targets)
         indices (np.ndarray, optional): Array of indices to the feature and adjacency matrix
             of the targets. Required if targets is not None.
+        corruption (bool): If True return shuffled features along with the regular features for Deep Graph
+            Infomax.
     """
 
     use_sparse = False
 
-    def __init__(self, features, A, targets=None, indices=None):
+    def __init__(self, features, A, targets=None, indices=None, corruption=False):
```

Comment:
It feels clunky to me to have to push down this corruption into each model manually, similar to it being clunky to have to duplicate GCN's construction with `GCNInfoMax`. I wonder if there's a way we can have have a wrapper like `CorruptingGenerator` around another Generator, similar to the proposal above for `InfoMax` to be a wrapper for `GCN`/`CorruptibleModel`?

Reply:
I've got a wrapper for the generator (and the model as well) that can handle all full batch models and I think can be easily generalized to other model types.

Reply:
Can we have a subclass `CorruptedFullBatchSequence(FullBatchSeqeuence)` that overrides the `__getitem__` method?

Reply:
The benefit of the wrapper is that it can handle corrupting multiple Sequences, currently it can handle `FullBatchNodeSequence`, `SparseFullBatchNodeSequence`, `RelationalFullBatchNodeSequence`. And it should be extendable to handle the GraphSage samplers as well. So in this case one class can handle the work of 3+ subclasses.

Also, users only access the `Sequence` objects through our generators, so we either have to subclass our generators or add extra arguments just to select whether to use a corrupted Sequence or not both of which feel clunky and end up duplicating code.